docs/design/CONFIG_API.md

```diff
@@ -0,0 +1,249 @@
+
+// Provider defines the configuration of the Kubernetes provider.
+type Provider struct {
+	// TODO: Define Kubernetes configuration fields, e.g. restrict namespaces to watch
+	//       Gateway/HTTPRoute resources.
+}
+```
+
+### Gateway
+Gateway defines desired configuration of [Gateway API][gw_api] controllers that reconcile and translate Gateway API
+resources into the Intermediate Representation (IR). Refer to the Envoy Gateway [design doc][design_doc] for additional
+details.
+
+### Providers
+Providers define desired configuration of Envoy Gateway providers. Providers are infrastructure components that Envoy
+Gateway calls to establish its runtime configuration. Providers are defined by a `+support` marker to indicate the
+stability of the provider. Refer to the Envoy Gateway [design doc][design_doc] for additional details.
+
+### XdsServer
+XdsServer defines desired configuration of [Xds][xds] server configuration parameters. Refer to the Envoy Gateway
+[design doc][design_doc] for additional details.
+
+### Configuration Example
+The configuration file is defined by the EnvoyGatewayConfigSpec API type. At startup, Envoy Gateway searches for the
+configuration at "/etc/envoy-gateway/config.yaml".
+
+Start Envoy Gateway:
+```shell
+envoy-gateway start
+```
+Since the configuration file does not exist, Envoy Gateway will start with default configuration parameters.
+
+The Kubernetes provider can be configured explicitly using `controlPlane.providers.kubernetes`:
+```yaml
+$ cat << EOF > /etc/envoy-gateway/config.yaml
+controlPlane:
+  providers:
+    kubernetes: {}
+EOF
+```
+This configuration will cause Envoy Gateway to use the Kubernetes provider with default configuration settings.
+
+The Kubernetes provider can be configured using the `Providers` API. For example, the `foo` field can be set to "bar":
+```yaml
+$ cat << EOF > /etc/envoy-gateway/config.yaml
+controlPlane:
+  providers:
+    kubernetes:
+      foo: bar
+EOF
+```
+__Note:__ The `Providers` API from the Kubernetes package is currently undefined and `foo: bar` is provided for
+illustration purposes only.
+
+The same API structure is followed for each supported provider. Multiple providers can be specified, for example:
+```yaml
+$ cat << EOF > /etc/envoy-gateway/config.yaml
+controlPlane:
+  providers:
+    kubernetes:
+      foo: bar
+    futureProvider:
+      foo: bar
+EOF
+```
+
+Envoy Gateway will parse providers to ensure only supported provider combinations are specified. If providers are valid,
+Envoy Gateway will merge the configuration from each provider into its IR. Envoy Gateway's Translator is then
+responsible for translating the IR into xDS resources that are pushed to managed data plane instances.
+
+__Note:__ To learn more about the Translator, refer to the Envoy Gateway [design doc][design_doc].
+
+As with Providers, Gateway API and xDS control plane services can be configured using `spec.gateway` and
+`spec.xdsServer`respectively. For example:
```

Comment:
Small nit, it should read `controlPlane.gateway` and `controlPlane.xdsServer`.

Reply:
Commit 9efda09 resolves this comment.